src/core/maps/grid_map.h

```diff
@@ -8,11 +8,18 @@
 #include "grid_cell_factory.h"
 #include "../geometry_utils.h"
 
+/**
+ * \brief Class describes Map that consists of Cells
+ */
 class GridMap {
 public:
   using Cell = std::shared_ptr<GridCell>;
 public:
   // TODO: cp, mv ctors, dtor
+  /**
+   * Constructor with parameters.
```

Comment:
Consider smth like that "Creates a grid map whose cells' type is defined by a given factory."